From d103eaeae169708ca567f092182a89b79e5ab9db Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 20 Oct 2024 07:52:33 -0700
Subject: [PATCH 1/2] Include linux/types.h for __u16/__u32/__u64 type

This header is included indirectly with glibc but when using musl
it ends up with compilation failure

BcachefsUtils.cc:85:20: error: use of undeclared identifier '__u32'
   85 |             args.dirfd = (__u32) fddst;
      |                           ^

Upstream-Status: Submitted [https://github.com/openSUSE/snapper/pull/945]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 snapper/BcachefsUtils.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/snapper/BcachefsUtils.cc b/snapper/BcachefsUtils.cc
index e9163ffb..1d328a78 100644
--- a/snapper/BcachefsUtils.cc
+++ b/snapper/BcachefsUtils.cc
@@ -24,6 +24,7 @@
 
 #include <cstring>
 #include <cerrno>
+#include <linux/types.h>
 #include <sys/stat.h>
 #include <sys/ioctl.h>
 
